### ISSUE TITLE
Generate a mock data set for a specific model in likelihood space.

### ISSUE THREAD

**Issue by Chris-Pedersen** [andreufont/LyaCosmoParams#12]
This will allow us to test the emulator error by seeing how well the input parameters are recovered. We can rescale the mean fluxes and temperatures to follow a particular model, and add the data covariance from BOSS.

**Comment by keirkwame** [comment]
If possible, it would be good to have a few of these reasonably distributed through the prior volume.

**Comment by andreufont** [comment]
Here is what I imagine:
 - choose a set of points in the prior volume, in the likelihood parameters space (still to be defined).
 - for each point, identify a set of simulation parameters that we believe should roughly reproduce the the likelihood parameters. This will be easier for cosmology, much harder for IGM parameters (see below).
 - we then run the simulation, and evaluate whether the IGM parameters at the different redshifts are close to what the likelihood parameters wanted. 
 - we could try to make it closer by running post-processing the snapshots, changing (mF, T_0, gamma), but we will not be able to post-process k_F.
 - if k_F is quite off, we could choose a different point in likelihood space that has a different k_F(z), and pretend that this is the one we wanted to test.

This would be easier if we had the function that we were talking about yeseterday, something like:
`def write_treecool_file(mf_model,T_model,kF_model) `
where ideally these models would be functions of z, not just a number.

But even if we had these functions, the calibration of the function will depend on cosmology, and probably also no box size / resolution / gadget version... but at least they should hopefully bring us to the right ball park.

**Comment by Chris-Pedersen** [comment]
Does it have to be done this way round? Couldn't we run a generic roughly-Planck cosmology and find the likelihood parameters of that? We could fit the IGM parameters, and just rescale the biggest outliers to the fit.

For this data set should we run a large box? Like a 1024**3 L=60, paired & fixed?

**Comment by andreufont** [comment]
Yes, we can start by doing what you suggest. But at some point it will be good to be able to make mocks for any hypothetical cosmology, to be able to show that we would recover the right cosmology even if the truth was far away from our fiducial (Planck-like) cosmology.

I think it would be great if you run the pair of simulations that you suggest, it would be good to run a first 1024 for this project, and it will allow many tests. We can discuss exactly what to run.

**Comment by Chris-Pedersen** [comment]
Continuing the discussion from #37 : for the mock data currently in the repo, using the current thermal models the fits and parameters are:

![thermal_parameters](https://user-images.githubusercontent.com/16047009/63771743-a4ca0800-c8cf-11e9-85cf-1c540cea02f6.png)

```
ln_tau_0 =  -1.03436530241446
ln_tau_1 =  3.6744666006830182
ln_kF_0 =  4.8005529736106425
ln_kF_1 =  -0.2593195555041545
ln_gamma_0 =  0.3295042060454974
ln_gamma_1 =  -0.2521703939255174
T0_1 =  0.13626544653787526
T0_2 =  9.546039892898634
T0_3 =  -1.2041429220366868
```

**Comment by andreufont** [comment]
Hi @Chris-Pedersen - a couple of comments: 
 - Are you sure the values of kF_kms in the plot above are correct? Shouldn't they be around 0.1-0.2 s/km? Could you check the units you are using?
 - When looking at the new thermal_model code, I found a bit confusing to use ln_T0_coeff to refer to three coefficients that did not correspond to a polynomail of ln(T_0). We could decide to use a different name for those, to make clear the distinction between these and the other coefficients (ln_gamma_coeff, ln_kF_coeff, ln_tau_coeff...)

**Comment by Chris-Pedersen** [comment]
I'll double check the kF units - For the T0 coefficients I removed the ln_ to make some distinction, but we can give them more explicit names if you like.

**Comment by Chris-Pedersen** [comment]
Sorry yes there was a typo in the conversion, I was *dkms_dMpc instead of dividing. The updated plot & fit values are:

![IGM_fits](https://user-images.githubusercontent.com/16047009/63847535-e3ba9500-c985-11e9-9293-030c75e442ac.png)


```
ln_tau_0 =  -1.03436530241446
ln_tau_1 =  3.6744666006830182
ln_kF_0 =  -0.8077668277205104
ln_kF_1 =  1.9001923998886694
ln_gamma_0 =  0.3295042060454974
ln_gamma_1 =  -0.2521703939255174
T0_1 =  0.13626544653787526
T0_2 =  9.546039892898634
T0_3 =  -1.2041429220366868
```

**Comment by Chris-Pedersen** [comment]
Fixed in #44
